### PR TITLE
Add Support for simple text file.

### PR DESCRIPTION
**Pull request by bipinmdr07**
## Description
Add Support for simple text file.

## Example TODO
```text
# 2022-12-20
- Work on adding text file :CODING: // adding tag with :text: format
- Add todo list :OTHER: // everything without tag will automatically treated as other
- Meeting with client :MEETING:

# 2022-12-21
- Work on adding org file :CODING: // adding tag with :text: format
- Add todo list for my party :OTHER: // everything without tag will automatically treated as other
- Meeting with the client :MEETING:
```

### CODE REVIEW

**Review by kritish-dhaubanjar — APPROVED** [review]

File: src/constants/regex.ts [hunk 1]
```diff
@@ -1 +1,3 @@
 export const ORG_TIME_STAMP = /^[A-Z][a-z]{2}\s\d{4}\s([A-Z][a-z]+)\s\d{2}/
+export const TXT_TIME_STAMP = /\d{4}\-[0-1]\d\-[0-3]\d/ // YYYY-MM-DD
+export const TODO_TAG = /:[a-zA-Z]+:/
```
Comment (sbimochan) [inline src/constants/regex.ts:3]: add new end of line?

File: src/utils/formatter.ts [hunk 2]
```diff
@@ -76,3 +76,31 @@ export const formatOrgLogs = (allTasks: any[]) => {
...
+
+export const formatTxtLogs = (allTasks: string[]) => {
+  let obj: Record<string, formattedTaskType[]> = {}
+  let currentHeading: string = ''
+
+  allTasks.forEach((task: string) => {
+    const trimmedTask = task.trim()
+
+    // check if line starts with `#` followed by date
+    if (trimmedTask[0] === '#' && TXT_TIME_STAMP.test(task)) {
```
Comment (sbimochan) [inline src/utils/formatter.ts]:
```suggestion
    if (trimmedTask.startsWith('#') && TXT_TIME_STAMP.test(task)) {
```

File: src/utils/formatter.ts [hunk 3]
```diff
@@ -76,3 +76,31 @@ export const formatOrgLogs = (allTasks: any[]) => {
...
+      currentHeading = trimmedTask.replace('# ', '')
+      if (!obj[currentHeading]) obj[currentHeading] = []
+      // check if tasks starts with `-` character ignore if line don't start with -
```
Comment (sbimochan) [inline src/utils/formatter.ts]:
```suggestion
      // check if tasks start with `-` character ignore if line doesn't start with -
```

File: README.md [hunk 4]
```diff
@@ -1,14 +1,36 @@
...
+```text
+# 2022-12-20
+- Work on adding text file :CODING: // adding tag with :TEXT: format
+- Add todo list :OTHER: // everything without tag will automatically treated as other
+- Meeting with client :MEETING:
+
+# 2022-12-21
+- Work on adding org file :CODING: // adding tag with :TEXT: format
+- Add todo list for my party :OTHER: // everything without tag will automatically treated as other
+- Meeting with the client :MEETING:
```
Comment (sbimochan) [inline README.md]:
```suggestion
# 2022-12-20
- Work on adding text file :CODING: // adding a tag with `:TEXT:` format
- Add todo list `:OTHER:` // everything without a tag will automatically be treated as other
- Meeting with the client `:MEETING:`

# 2022-12-21
- Work on adding org file `:CODING:` // adding a tag with `:TEXT:` format
- Add todo list for my party `:OTHER:` // everything without a tag will automatically be treated as other
- Meeting with the client `:MEETING:`
```

File: src/utils/formatter.ts [hunk 5]
```diff
@@ -76,3 +76,31 @@ export const formatOrgLogs = (allTasks: any[]) => {
...
+    } else if (currentHeading && task.length && /^\-/.test(trimmedTask)) {
```
Comment (sbimochan) [inline src/utils/formatter.ts]: name and export that regex

File: src/utils/formatter.ts [hunk 6]
```diff
@@ -76,3 +76,31 @@ export const formatOrgLogs = (allTasks: any[]) => {
 
   return obj
 }
+
+export const formatTxtLogs = (allTasks: string[]) => {
+  let obj: Record<string, formattedTaskType[]> = {}
```
Comment (sbimochan) [inline src/utils/formatter.ts]: can this be named to record instead? and the function to know that it returns record as type of Object?

File: src/utils/formatter.ts (hunk 2 above)
Comment (sbimochan) [reply src/utils/formatter.ts]: however it's an optional to change to this because we don't need any other complex checks

**Pull request merged**